src/convert.rs

```diff
@@ -13,7 +13,12 @@ impl Into<HeapType> for IntoHeapType {
         match self.0 {
             wasmparser::HeapType::Func => HeapType::Func,
             wasmparser::HeapType::Extern => HeapType::Extern,
-            wasmparser::HeapType::Indexed(index) => HeapType::Indexed(index.into()),
+            wasmparser::HeapType::Concrete(wasmparser::UnpackedIndex::Module(index)) => {
+                HeapType::Concrete(index)
+            }
+            wasmparser::HeapType::Concrete(
+                wasmparser::UnpackedIndex::RecGroup(_) | wasmparser::UnpackedIndex::Id(_),
+            ) => todo!(),
```

Comment:
What's this about?

Reply:
There's not an obvious, 1-to-1 correspondence between https://docs.rs/wasmparser/latest/wasmparser/enum.HeapType.html and https://docs.rs/wasm-encoder/latest/wasm_encoder/enum.HeapType.html, so I guessed :shrug: .  I _think_ `HeapType::Concrete` is only used as part of WasmGC anyway, so it shouldn't matter to us.

Reply:
Can we make this a `panic` with an appropriate message?

Reply:
Done.  We could probably just not support `RefType`s or `HeapType`s at all, since they aren't used by any known SDK, but it was easy enough to support at least the variants that had a clear mapping from the `wasmparser` version to the `wasm-encoder` version.